File: netlify/functions/download.js
// netlify/functions/download.js
const { createClient } = require('@supabase/supabase-js');

const supabase = createClient(
  process.env.SUPABASE_URL,
  process.env.SUPABASE_ANON_KEY
);

exports.handler = async (event, context) => {
  if (event.httpMethod === 'OPTIONS') {
    return {
      statusCode: 200,
      headers: {
        'Access-Control-Allow-Origin': '*',
        'Access-Control-Allow-Headers': 'Content-Type',
        'Access-Control-Allow-Methods': 'GET, OPTIONS'
      }
    };
  }

  try {
    const { fileId } = event.queryStringParameters || {};
    
    if (!fileId) {
      return {
        statusCode: 400,
        body: JSON.stringify({ error: 'Missing fileId parameter' })
      };
    }

    const { data, error } = await supabase
      .from('files')
      .select('*')
      .eq('id', fileId)
      .single();

    if (error || !data) {
      return {
        statusCode: 404,
        body: JSON.stringify({ error: 'File not found' })
      };
    }

    // Track download event
    await supabase.from('analytics').insert({
      event_type: 'download',
      file_id: fileId,
      file_name: data.file_name,
      file_size: data.file_size,
      action: 'file_downloaded',
      user_ip: event.headers['x-forwarded-for'] || 'unknown'
    });

    return {
      statusCode: 200,
      headers: {
        'Content-Type': 'application/json',
        'Access-Control-Allow-Origin': '*'
      },
      body: JSON.stringify({
        fileName: data.file_name,
        fileData: data.file_url,
        contentType: data.content_type
      })
    };
  } catch (error) {
    console.error('Download error:', error);
    return {
      statusCode: 500,
      headers: { 'Access-Control-Allow-Origin': '*' },
      body: JSON.stringify({ error: error.message })
    };
  }
};

File: netlify/functions/download.py
import json
import base64
from datetime import datetime

def handler(event, context):
    """
    Generiše download link za fajl
    """
    # Proveri HTTP metodu
    if event['httpMethod'] != 'GET':
        return {
            'statusCode': 405,
            'headers': {
                'Content-Type': 'application/json',
                'Access-Control-Allow-Origin': '*'
            },
            'body': json.dumps({'error': 'Method not allowed'})
        }
    
    try:
        # Uzmi fileId iz query parametara
        query_params = event.get('queryStringParameters', {})
        file_id = query_params.get('fileId')
        
        if not file_id:
            return {
                'statusCode': 400,
                'headers': {
                    'Content-Type': 'application/json',
                    'Access-Control-Allow-Origin': '*'
                },
                'body': json.dumps({'error': 'Missing fileId parameter'})
            }
        
        # Simulacija - u produkciji bi učitao pravi fajl
        mock_content = f"This is sample content for file {file_id}\nDownloaded at: {datetime.now()}"
        encoded_content = base64.b64encode(mock_content.encode()).decode()
        
        return {
            'statusCode': 200,
            'headers': {
                'Content-Type': 'application/json',
                'Access-Control-Allow-Origin': '*',
                'Access-Control-Allow-Headers': 'Content-Type'
            },
            'body': json.dumps({
                'fileId': file_id,
                'fileName': f'file-{file_id}.txt',
                'fileData': f'data:text/plain;base64,{encoded_content}',
                'downloadUrl': f'/download/{file_id}'  # Simulirani URL
            })
        }
        
    except Exception as e:
        return {
            'statusCode': 500,
            'headers': {
                'Content-Type': 'application/json',
                'Access-Control-Allow-Origin': '*'
            },
            'body': json.dumps({
                'error': 'Download failed',
                'details': str(e)
            })
        }